#En este notebook aplicaremos las diferentes etapas del pre procesamiento:



In [2]:
import numpy as np
import pandas as pd
%matplotlib inline

Definición del problem

Contexto
Según la Organización Mundial de la Salud (OMS), el accidente cerebrovascular es la 2ª causa de muerte a nivel mundial, responsable de aproximadamente el 11% del total de muertes. Este conjunto de datos se utiliza para predecir si es probable que un paciente sume un accidente cerebrovascular en función de los parámetros de entrada como el sexo, la edad, varias enfermedades y el estado de tabaquismo. Cada fila de los datos proporciona información relavante sobre el paciente.

Información de atributos

1) id: identificador único\
2) género: "Masculino", "Femenino" u "Otro"\
3) edad: edad del paciente\
4) hipertensión: 0 si el paciente no tiene hipertensión, 1 si el paciente tiene hipertensión\
5) heart_disease: 0 si el paciente no tiene ninguna enfermedad cardíaca, 1 si el paciente tiene una enfermedad cardíaca\
6) ever_married: "No" o "Sí"\
7) work_type: "hijos", "Govt_jov", "Never_worked", "privado" o "trabajador por cuenta propia"\
8) Residence_type: "Rural" o "Urbano"\
9) avg_glucose_level: nivel promedio de glucosa en sangre\
10) BMI: índice de masa corporal\
11) smoking_status: "anteriormente fumado", "nunca fumado", "fuma" o "Desconocido"*\
12) accidente cerebrovascular: 1 si el paciente tuvo un accidente cerebrovascular o 0 si no











In [4]:
# Reemplaza el nombre por el de tu archivo real
df = pd.read_csv("healthcare-dataset-stroke-data.csv")

omar.salinas@unab.cl

Información del DataSet preliminar. Lo estamos revisando

In [11]:
df

,id,gender,age,hypertension,heart_disease,ever_married,work_type,Residence_type,avg_glucose_level,bmi,smoking_status,stroke
0,9046,Male,67.0,0,1,Yes,Private,Urban,228.69,36.6,formerly smoked,1
1,51676,Female,61.0,0,0,Yes,Self-employed,Rural,202.21,NaN,never smoked,1
2,31112,Male,80.0,0,1,Yes,Private,Rural,105.92,32.5,never smoked,1
3,60182,Female,49.0,0,0,Yes,Private,Urban,171.23,34.4,smokes,1
4,1665,Female,79.0,1,0,Yes,Self-employed,Rural,174.12,24.0,never smoked,1
...,...,...,...,...,...,...,...,...,...,...,...,...
5105,18234,Female,80.0,1,0,Yes,Private,Urban,83.75,NaN,never smoked,0
5106,44873,Female,81.0,0,0,Yes,Self-employed,Urban,125.20,40.0,never smoked,0
5107,19723,Female,35.0,0,0,Yes,Self-employed,Rural,82.99,30.6,never smoked,0
5108,37544,Male,51.0,0,0,Yes,Private,Rural,166.29,25.6,formerly smoked,0


In [12]:
df.shape

(5110, 12)

El DataSet contine 12 atributos o variables (columnas) y 5110 registros (filas).

In [18]:
df.sample(100)

,id,gender,age,hypertension,heart_disease,ever_married,work_type,Residence_type,avg_glucose_level,bmi,smoking_status,stroke
1631,45754,Female,20.00,0,0,No,Private,Urban,75.94,28.3,never smoked,0
2436,22259,Male,10.00,0,0,No,children,Rural,77.51,21.9,Unknown,0
182,39912,Female,32.00,0,0,Yes,Private,Rural,76.13,29.9,smokes,1
3888,29201,Male,1.56,0,0,No,children,Rural,109.12,18.9,Unknown,0
732,31308,Female,49.00,0,0,Yes,Private,Urban,114.50,35.9,formerly smoked,0
...,...,...,...,...,...,...,...,...,...,...,...,...
3120,26777,Male,22.00,0,0,No,Private,Rural,86.53,20.8,never smoked,0
3241,41593,Female,76.00,0,0,Yes,Self-employed,Rural,70.29,33.4,formerly smoked,0
1669,1842,Male,58.00,0,0,Yes,Private,Urban,94.00,NaN,Unknown,0
148,72081,Female,57.00,1,0,Yes,Govt_job,Rural,67.41,32.9,never smoked,1


Revisar la contidad de valores nulos del DataFrame, el atributo bmi tiene valores nulos o missing values.


In [19]:
df.isnull().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64